### PR TITLE
[WIP] Support hidden fields

### PR DESCRIPTION
**Pull request by ayanai1**
- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes

### CODE REVIEW

File: src/main/java/net/sf/jabref/model/entry/BibEntry.java [hunk 1]
```diff
@@ -24,24 +24,16 @@
...
-import java.util.Date;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Objects;
-import java.util.Optional;
-import java.util.Set;
-import java.util.TreeSet;
+import java.util.*;
```
Comment (koppor) [inline src/main/java/net/sf/jabref/model/entry/BibEntry.java]: Please update your Eclipse settings. See https://github.com/JabRef/jabref/issues/1091

File: src/main/java/net/sf/jabref/model/entry/BibEntry.java [hunk 2]
```diff
@@ -600,30 +631,4 @@ public void addKeywords(List<String> keywords) {
...
-    public Collection<String> getFieldValues() {
-        return fields.values();
-    }
-
-    public Map<String, String> getFieldMap() {
-        // TODO Auto-generated method stub
-        return fields;
-    }
-
-    @Override
```
Comment (koppor) [inline src/main/java/net/sf/jabref/model/entry/BibEntry.java]: Why is this method removed?

File: src/main/java/net/sf/jabref/model/entry/BibEntry.java [hunk 3]
```diff
@@ -506,7 +535,7 @@ public String getPublicationDate() {
             return null;
         }
 
-        String year = getField("year");
+        String year = YearUtil.toFourDigitYear(getField("year"));
```
Comment (koppor) [inline src/main/java/net/sf/jabref/model/entry/BibEntry.java]: Do not update data value without notifying the user. If you want to propose such a change, please open a new PR.

File: src/main/java/net/sf/jabref/model/entry/BibEntry.java [hunk 4]
```diff
@@ -548,11 +577,13 @@ public void putKeywords(List<String> keywords) {
         if (newValue == null) {
             if (oldValue != null) {
                 this.clearField("keywords");
+                changed = true;
```
Comment (koppor) [inline src/main/java/net/sf/jabref/model/entry/BibEntry.java]: Do not readd these things. See https://github.com/JabRef/jabref/pull/1297

File: src/main/java/net/sf/jabref/model/entry/BibEntry.java [hunk 5]
```diff
@@ -548,11 +577,13 @@ public void putKeywords(List<String> keywords) {
...
             }
             return;
         }
         if ((oldValue == null) || !oldValue.equals(newValue)) {
             this.setField("keywords", newValue);
+            changed = true;
```
Comment (koppor) [inline src/main/java/net/sf/jabref/model/entry/BibEntry.java]: see above

File: src/main/java/net/sf/jabref/model/entry/BibEntry.java [hunk 6]
```diff
@@ -379,6 +372,42 @@ public void clearField(String name) {
     }
 
     /**
+     * Hide single chosen optional Field
+     * @param name field name has to be checked
+     */
+    public void toggleOptionalFieldVisibility(String name) {
```
Comment (koppor) [inline src/main/java/net/sf/jabref/model/entry/BibEntry.java]: Unfortunately, this is not what is requested at https://github.com/JabRef/jabref/issues/759

I want to select a single field in the UI to hide or unhide (second bullet point).

The third bullet point means that I want to unhide all fields in the cleanup dialog. This is an request, which should come at last, the coloring and the hide/unhide functionality is more important.

File: src/main/java/net/sf/jabref/model/entry/BibEntry.java [hunk 7]
```diff
@@ -326,12 +323,7 @@ public void setField(String name, String value) {
         Objects.requireNonNull(name, "field name must not be null");
         Objects.requireNonNull(value, "field value must not be null");
 
-        if (value.isEmpty()) {
```
Comment (koppor) [inline src/main/java/net/sf/jabref/model/entry/BibEntry.java]: Why is this code removed? See https://github.com/JabRef/jabref/pull/1281#discussion_r60948627 for a discussion about the addition of this code.

**Comment by matthiasgeiger** [comment]
regarding the comments from @koppor looks like a merge conflict was "solved" by "Force ours"...

@ayanai1 This is a "quick" but in most cases not a valid solution to resolve conflicts... The changes you reverted were made intentionally!

**Comment by stefan-kolb** [comment]
What is the current status here?

**Comment by Braunch** [comment]
We are looking forward to our next programming-marathon on Tuesday the 19th. This PR is on the agenda for that day.

**Comment by tschechlovdev** [comment]
Could you please push the changes you make, so that we can see the progression?

**Comment by koppor** [comment]
Follow up at #1655

**Pull request closed without merging**